### PR TITLE
Add check for PVC status in E2E tests

### PR DESCRIPTION
**Pull request by dsimansk**
## Description

Not every cluster might be ready to fulfill PVC. 

## Changes

* :broom: Add check for PVC status in E2E tests

## Reference

Fixes #

**Release Note**

```release-note
NONE
```

### CODE REVIEW

**Comment by dsimansk** [comment]
PTAL, it's related to errors in OpenShift executed tests.
/cc @vyasgun

File: test/e2e/service_test.go [hunk 1]
```diff
@@ -240,11 +240,17 @@ func serviceCreateWithMount(r *test.KnRunResultCollector) {
 	assert.NilError(r.T(), err)
 	r.AssertNoError(out)
 
-	r.T().Log("update service with a new pvc mount")
-	out = r.KnTest().Kn().Run("service", "update", "test-svc", "--mount", "/mydir5=pvc:test-pvc")
-	r.AssertNoError(out)
+	_, err = kubectl.Run("wait", "--for=condition=Ready", "pvc/test-pvc", "--timeout=30s")
```
Comment (vyasgun) [inline test/e2e/service_test.go:243]: The condition check will not work in the case of PVC as it doesn't have the field. Checking for `phase == Bound` should work instead?
```suggestion
	_, err = kubectl.Run("wait", "--for='jsonpath={..status.phase}'=Bound", "pvc/test-pvc", "--timeout=30s")
```

Comment (dsimansk) [reply test/e2e/service_test.go]: Yep, that's better thanks. I should have put a temp `.fail()` there first to test it out.

**Comment by vyasgun** [comment]
Thanks!

/lgtm

**Pull request merged**